### DISCUSSION TITLE
How pydantic handles if request body is None

### DISCUSSION THREAD

**Discussion by ajay-sudosu** [fastapi/fastapi#11559] [Questions]
### Privileged issue

- [X] I'm @tiangolo or he asked me directly to create an issue here.

### Issue Content

Suppose I have a model:
class User(Basemodel):
    name: str
    age: str
    
And I have an endpoint something like:
@app.post("/user", description="User add")
def add_user(add_user_data: User):
    so_something()
    
 I am expecting a request body with the user details as show above in pydantic model
 Question is how should I handle the case when request body is None 
 
 Note:
 I am using fastapi add_exception_handler() function to handle the incoming request before reaching it to the endpoint some thing like:
 
 def handle_pydantic_validations(request: Request, exc: ValidationError):
    errors = exc.args[0]
    error_list = [{error["loc"][1]: error["msg"]} for error in errors]
    return make_response(status_code=status.HTTP_400_BAD_REQUEST, body={"body": error_list,
                                                                        "status_code": status.HTTP_400_BAD_REQUEST})

**Comment by YuriiMotov** [comment]
Hi!
Could you?:
1) format your question (use "code" block to format code examples)
2) provide minimal example of what you have (but including imports and so on, so that we can just run it)
3) explain more clearly what you want to achieve

**Comment by ajay-sudosu** [comment]
I am expecting a request body as Items (Pydantic model). Also I am using an exception handler before the request reaches to my endpoint I want to check if request body is None or not how can I achieve this because exception handler is not catching is by itself. 

`from fastapi import FastAPI, Request`
`from fastapi.responses import JSONResponse`
`from fastapi.exceptions import RequestValidationError`
`from pydantic import BaseModel`
`from typing import Union`

`app = FastAPI()`


`class Item(BaseModel):`
    `name: str`
    `description: Union[str, None] = None`
    `price: float`
    `tax: Union[float, None] = None`


`@app.post("/items/")`
`async def create_item(item: Item):`
    `return {"item": item}`


`@app.exception_handler(RequestValidationError)`
`async def validation_exception_handler(request: Request, exc: RequestValidationError):`
    `return JSONResponse(
        status_code=400,
        content={
            "detail": exc.errors(),
            "body": exc.body
        }
    )`


`@app.get("/")`
`async def root():`
    `return {"message": "Welcome to the FastAPI app!"}`

**Reply by YuriiMotov** [reply]
Still not sure I understand your question. But if your question is "how should I handle the case when request body is None" then the answer is:
You don't need to handle this situation manually.
FastAPI will raise `RequestValidationError` if your endpoint has a required parameter (`item: Item` in your case) and there are no data in request body. This will couse FastAPI to return `422` response by default. But since you set `exception_handler` for `RequestValidationError` it will be changed to `400` response code.
